### DISCUSSION TITLE
GraphicsDevice Public and Private Properties

### DISCUSSION THREAD

**Discussion by epreston** [playcanvas/engine#5640] [General]
In this PR, a number of methods where marked as no longer exposed as public in derived classes.

https://github.com/playcanvas/engine/pull/4013

Is this still the case ?

Specifically:

```
GraphicsDevice.setIndexBuffer
GraphicsDevice.setVertexBuffer
```

At the moment, they are not marked as "private" so typedocs is trying to process a few of them.

**Comment by mvaligursky** [comment]
These two methods are marked as public currently: setIndexBuffer, setVertexBuffer.
So I'm not entirely sure what you are asking here, could you please clarify?

**Reply by epreston** [reply]
I'm preparing updates for published typedoc documentation.  

It fails to create documentation for these with warnings. I did a quick check to determine if it should be documenting them in the first place.  I found this PR which suggested they should get an `@ignore` to stop them from being added to published documentation.

**Reply by mvaligursky** [reply]
They seem to documented correctly to me here: https://github.com/playcanvas/engine/blob/bdf8eac6a46ba1eb0558f65278b0bbb02f094745/src/platform/graphics/graphics-device.js#L520-L546

What warnings are you getting?

**Reply by epreston** [reply]
The documentation comment links to a `draw` method which does not exist on the `GraphicsDevice` base class.  

In another thread, we talked about creating the abstract method to satisfy this.  I'll update that.

Once fixed, it will appear in public documentation. I want to be sure it should be published.
